benchmarks/grpo/evaluate/humaneval_eval.py

```
@@ -0,0 +1,75 @@
+                break
+            if line.strip().startswith("class "):
+                break
+    return '\n'.join(code_lines)
+
+def test_code(code: str, test: str) -> bool:
+    """Execute code + test in isolated environment."""
+    with tempfile.NamedTemporaryFile(mode='w', suffix='.py', delete=False) as f:
+        f.write(code + '\n' + test)
+        f.flush()
+        try:
+            result = subprocess.run(
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)